### PR TITLE
Cleanup/Implement some todos

### PR DESCRIPTION
**Pull request by ichttt**
Simple PR, cleans up 4 Todos by implementing the code:
- WorldWorkerManager
- BlockSnapshot
- SelectiveReloadStateHandler
- ConsoleCommandCompleter
- DefaultPermissionHandler

### CODE REVIEW

**Review by tterrag1098 — APPROVED** [review]

File: src/main/java/net/minecraftforge/server/permission/IPermissionHandler.java [hunk 1]
```diff
@@ -38,9 +39,9 @@
     Collection<String> getRegisteredNodes();
 
     /**
-     * Use {@link PermissionAPI#hasPermission(GameProfile, String, IContext)}
+     * Use {@link PermissionAPI#hasPermission(MinecraftServer, GameProfile, String, IContext)}
      */
-    boolean hasPermission(GameProfile profile, String node, @Nullable IContext context);
+    boolean hasPermission(@Nullable MinecraftServer server, GameProfile profile, String node, @Nullable IContext context);
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/server/permission/IPermissionHandler.java]: Why did you change the API here? I understand the old TODO called some helper method to get the current server. So it would be better to get the server internally instead of expecting everyone to send it in.

**Comment by LexManos** [comment]
This is fine, execpt for the API change in hasPermissions.

File: src/main/java/net/minecraftforge/server/permission/IPermissionHandler.java (hunk 1 above)
Comment (ichttt) [reply src/main/java/net/minecraftforge/server/permission/IPermissionHandler.java]: I did not know that there was a replacement function for the static server instance.
